### PR TITLE
Issue #1404: Implemented tests for CloudRegionApiService

### PR DESCRIPTION
**Pull request by youkofan**
Implement tests for ACL layer (for **region package**) - https://github.com/epam/cloud-pipeline/issues/1404
Implemented tests for CloudRegionApiService.
Signed-off-by: Iurii_Kofanov <youkofan@gmail.com>

### CODE REVIEW

File: api/src/test/java/com/epam/pipeline/test/acl/AclTestBeans.java [hunk 1]
```diff
@@ -63,20 +91,26 @@
...
+    protected PermissionFactory permissionFactory;
+
     @Autowired
     protected PermissionEvaluator permissionEvaluator;
 
     @Autowired
     protected MessageHelper messageHelper;
 
+    @Autowired
+    protected PermissionsService mockPermissionsService;
```
Comment (ekazachkova) [inline api/src/test/java/com/epam/pipeline/test/acl/AclTestBeans.java:112]: Is it mock?

File: api/src/test/java/com/epam/pipeline/test/acl/AclTestBeans.java [hunk 2]
```diff
@@ -212,6 +234,98 @@
...
+    @Bean
+    protected JsonService mockJsonService() {
+        return Mockito.mock(JsonService.class);
+    }
+
+    @MockBean
+    protected FolderDao mockFolderDao;
+
+    @Bean
+    protected DataStorageManager mockDataStorageManager() { //todo
```
Comment (ekazachkova) [inline api/src/test/java/com/epam/pipeline/test/acl/AclTestBeans.java]: Could you please cleanup code?

File: api/src/test/java/com/epam/pipeline/acl/region/CloudRegionApiServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+
+public class CloudRegionApiServiceTest extends AbstractAclTest {
+
+    @Autowired
+    private CloudRegionApiService cloudRegionApiService;
+
+    @Autowired
+    private CloudRegionManager cloudRegionManager;
```
Comment (ekazachkova) [inline api/src/test/java/com/epam/pipeline/acl/region/CloudRegionApiServiceTest.java:50]: Could you please rename this field to `mockCloudRegionManager`?

File: api/src/test/java/com/epam/pipeline/acl/region/CloudRegionApiServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    @Test
+    @WithMockUser(username = SIMPLE_USER_ROLE)
+    public void shouldReturnAllCloudRegionsWhenPermissionIsGranted() {
+        initAclEntity(region,
+                Collections.singletonList(new UserPermission(SIMPLE_USER_ROLE, AclPermission.READ.getMask())));
+        doReturn(clouds).when(cloudRegionManager).loadAll();
+
+        List<? extends AbstractCloudRegion> abstractCloudRegions = cloudRegionApiService.loadAll();
+
+        System.out.println("abstractCloudRegions: " + abstractCloudRegions.get(0).getMask());
```
Comment (ekazachkova) [inline api/src/test/java/com/epam/pipeline/acl/region/CloudRegionApiServiceTest.java:184]: Could you please cleanup here?

File: api/src/test/java/com/epam/pipeline/acl/region/CloudRegionApiServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+
+        cloudRegionApiService.loadAllAvailable(CloudProvider.AWS);
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldReturnAllCloudRegionsForAdmin() {
+        doReturn(clouds).when(cloudRegionManager).loadAll();
+
+        List<? extends AbstractCloudRegion> abstractCloudRegions = cloudRegionApiService.loadAll();
```
Comment (ekazachkova) [inline api/src/test/java/com/epam/pipeline/acl/region/CloudRegionApiServiceTest.java:169]: Don't forget to make fields final. Please, correct here and elsewhere.

File: api/src/test/java/com/epam/pipeline/app/AclTestConfiguration.java [hunk 6]
```diff
@@ -144,6 +146,12 @@
     @MockBean
     protected LogManager mockLogManager;
 
+    @MockBean
```
Comment (ekazachkova) [inline api/src/test/java/com/epam/pipeline/app/AclTestConfiguration.java:149]: Do we really need this mocks here?

File: api/src/test/java/com/epam/pipeline/acl/region/CloudRegionApiServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+    @WithMockUser(username = SIMPLE_USER_ROLE)
+    public void shouldReturnCloudRegionWhenPermissionIsGranted() {
+        initAclEntity(region,
+                Collections.singletonList(new UserPermission(SIMPLE_USER_ROLE, AclPermission.READ.getMask())));
+        when(cloudRegionManager.load(eq(region.getId()))).thenReturn(region);
+
+        AbstractCloudRegion result = cloudRegionApiService.load(region.getId());
+
+        assertThat(result).isEqualTo(region);
+        assertThat(result.getId()).isEqualTo(region.getId());
```
Comment (ekazachkova) [inline api/src/test/java/com/epam/pipeline/acl/region/CloudRegionApiServiceTest.java:224]: Why do we need such checks?

File: api/src/test/java/com/epam/pipeline/acl/region/CloudRegionApiServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldReturnCloudRegionForAdmin() {
+        doReturn(region).when(cloudRegionManager).load(region.getId());
+
+        AbstractCloudRegion load = cloudRegionApiService.load(region.getId());
+
+        assertThat(load).isEqualTo(region);
+        assertThat(load.getId()).isEqualTo(clouds.get(0).getId());
```
Comment (ekazachkova) [inline api/src/test/java/com/epam/pipeline/acl/region/CloudRegionApiServiceTest.java:211]: Why do we need such checks?

**Review by ekazachkova — CHANGES_REQUESTED** [review]

File: api/src/test/java/com/epam/pipeline/test/acl/AclTestBeans.java (hunk 1 above)
Comment (youkofan) [reply api/src/test/java/com/epam/pipeline/test/acl/AclTestBeans.java:112]: No, this is not a mock. When this field was marked with @MockBean annotation, the tests did not work. I will change the name to "permissionService".

File: api/src/test/java/com/epam/pipeline/acl/region/CloudRegionApiServiceTest.java (hunk 8 above)
Comment (youkofan) [reply api/src/test/java/com/epam/pipeline/acl/region/CloudRegionApiServiceTest.java:211]: It seemed to me that an additional check would not be superfluous, but perhaps you are right. I will remove it.

File: api/src/test/java/com/epam/pipeline/acl/region/CloudRegionApiServiceTest.java (hunk 7 above)
Comment (youkofan) [reply api/src/test/java/com/epam/pipeline/acl/region/CloudRegionApiServiceTest.java:224]: It seemed to me that an additional check would not be superfluous, but perhaps you are right. I will remove it.

File: api/src/test/java/com/epam/pipeline/app/AclTestConfiguration.java (hunk 6 above)
Comment (youkofan) [reply api/src/test/java/com/epam/pipeline/app/AclTestConfiguration.java:149]: You're right. This mock is redundant here. Removed.

**Pull request closed without merging**